### PR TITLE
fix(rest): normalize type names in OpenAPI schema lookup

### LINKED ISSUE #2440: Bug: OpenAPI schema lookup fails for fully-qualified type paths
## Prerequisites

- [x] I have searched existing issues to verify this bug hasn't been reported

## Bug Description

The OpenAPI schema lookup in `EndpointInspector::create_request_body()` fails to match schemas when the `request_body_type` from endpoint metadata contains a fully-qualified type path.

Schemas are registered in the global registry with their short identifier name (e.g., `"CreateUserRequest"`), but the endpoint extraction via `quote!().to_string()` produces a path-qualified name (e.g., `"crate :: models :: CreateUserRequest"`).

This key mismatch causes the lookup at `get_all_schemas().get(body_type)` to always miss, falling back to `ObjectBuilder::new()` which produces an empty `{}` request body schema.

## Steps to Reproduce

1. Define a request body type in a submodule (e.g., `crate::models::CreateUserRequest`)
2. Register it with `#[derive(Schema)]` (registers as `"CreateUserRequest"`)
3. Use it in an endpoint decorator macro (produces `"crate :: models :: CreateUserRequest"` via `quote!`)
4. Generate OpenAPI spec
5. Observe that the request body schema is an empty object instead of the registered schema

## Expected Behavior

The schema lookup should normalize the fully-qualified type path by extracting the last path segment (e.g., `"CreateUserRequest"` from `"crate :: models :: CreateUserRequest"`), then match against the registry.

## Actual Behavior

The lookup uses the raw fully-qualified path string as the key, which never matches the short name in the registry. This causes all request bodies to fall back to empty `ObjectBuilder::new()` schemas.

## Environment

- Rust version: 1.87.0
- OS: macOS 15.4
- Crate version: reinhardt-rest (workspace)

## Additional Context

The fix should add a `normalize_type_name()` helper that extracts the last `::` segment from the type string, and apply it in `create_request_body()` before the registry lookup.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### PR DESCRIPTION
**Pull request by kent8192**
## Summary

- Add `normalize_type_name()` helper to extract the last path segment from fully-qualified type paths
- Fix schema registry lookup in `EndpointInspector::create_request_body()` to normalize type names before lookup
- Add unit tests for the normalization function and qualified path handling

## Type of Change

- [x] Bug fix (non-breaking change that fixes an issue)

## Motivation and Context

When endpoint metadata is extracted via `quote!().to_string()`, the `request_body_type` field contains fully-qualified paths like `"crate :: models :: CreateUserRequest"`. However, schemas are registered in the global registry with their short identifier name (e.g., `"CreateUserRequest"`). This key mismatch causes the lookup to always fail, falling back to an empty `ObjectBuilder::new()` schema — resulting in empty `{}` request bodies in the generated OpenAPI spec.

Fixes #2440

## How Was This Tested?

- `cargo nextest run --package reinhardt-rest` — all 608 tests pass
- New unit tests added:
  - `test_normalize_type_name_simple` — simple name passthrough
  - `test_normalize_type_name_fully_qualified` — path with spaces from `quote!()`
  - `test_normalize_type_name_compact_path` — path without spaces
  - `test_normalize_type_name_single_segment_with_colons` — two-segment path
  - `test_normalize_type_name_empty_string` — edge case
  - `test_create_request_body_with_qualified_path` — end-to-end test with qualified type name

## Checklist

- [x] I have followed the [Contributing Guidelines](../blob/main/CONTRIBUTING.md)
- [x] I have followed the [Commit Guidelines](../blob/main/instructions/COMMIT_GUIDELINE.md)
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have formatted the code with `cargo make fmt-fix`
- [x] I have checked the code with `cargo make clippy-check`
- [ ] I use self-hosted runner for CI (Repository owner only)

## Related Issues

- Fixes #2440

### Type Label (select one)
- [x] `bug` - Bug fix

### Scope Label (select all that apply)
- [x] `api` - REST API, serializers, views

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: crates/reinhardt-rest/src/openapi/endpoint_inspector.rs [hunk 1]
```diff
@@ -36,6 +36,19 @@ impl Default for InspectorConfig {
...
+///
+/// Handles fully-qualified paths like `"crate :: models :: CreateUserRequest"`
+/// (produced by `quote!().to_string()`) by returning just `"CreateUserRequest"`.
+/// Simple names without `::` are returned as-is.
+fn normalize_type_name(type_str: &str) -> &str {
+	type_str
+		.rsplit("::")
+		.next()
+		.map(|s| s.trim())
+		.unwrap_or(type_str)
```
[the remark this answers is not in the document]
Comment (kent8192) [reply crates/reinhardt-rest/src/openapi/endpoint_inspector.rs]: Fixed. Simplified `normalize_type_name` to use `rsplit_once("::")` which makes the branching explicit:
- When `::` is found, return the last segment trimmed
- When no `::` is found, return the input trimmed

The doc comment now explicitly documents the trimming behavior for both cases.

See commit 30d4aa2fe.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

File: crates/reinhardt-rest/src/openapi/endpoint_inspector.rs [hunk 2]
```diff
@@ -514,4 +532,93 @@ mod tests {
...
+					description
+				);
+				assert!(
+					!description.contains("crate ::"),
+					"Description should not contain path qualifiers, got: {}",
+					description
+				);
+			}
+		}
+	}
```
[the remark this answers is not in the document]
Comment (kent8192) [reply crates/reinhardt-rest/src/openapi/endpoint_inspector.rs]: Fixed. The test now:
1. Registers a test schema via `inventory::submit!` with known properties (title, description, `test_field` property)
2. Asserts the schema is present in `get_all_schemas()` before testing
3. Uses a fully-qualified path (`crate :: models :: QualifiedPathTestSchema`) that normalizes to the registered name
4. Asserts the returned schema has the registered title, description, and properties -- proving the registry lookup succeeded rather than falling back to a placeholder
5. Uses explicit `match` arms instead of nested `if let`, so the test will fail (not silently pass) if the schema is a `$ref` or unexpected variant

See commit 30d4aa2fe.

🤖 Generated with [Claude Code](https://claude.com/claude-code)
